internal/pkg/expanders/containerApp.go

```diff
@@ -0,0 +1,286 @@
+				"subscriptionId":       currentItem.Metadata["subscriptionId"],
+				"resourceGroupName":    currentItem.Metadata["resourceGroupName"],
+				"containerAppName":     currentItem.Metadata["containerAppName"],
+				"revisionName":         currentItem.Metadata["revisionName"],
+				"replicaName":          currentItem.Metadata["replicaName"],
+			},
+		})
+	}
+	return ExpanderResult{
+		IsPrimaryResponse: true,
+		Nodes:             newItems,
+		Response:          ExpanderResponse{Response: string(containersDisplay), ResponseType: interfaces.ResponseJSON},
+		SourceDescription: "ContainerAppExpander request",
+	}
+}
+func (e *ContainerAppExpander) expandReplicaContainer(ctx context.Context, currentItem *TreeNode) ExpanderResult {
+	return ExpanderResult{
+		IsPrimaryResponse: true,
+		Nodes:             []*TreeNode{},
+		Response:          ExpanderResponse{Response: "Use the command panel to show container logs", ResponseType: interfaces.ResponsePlainText},
+		SourceDescription: "ContainerAppExpander request",
+	}
+}
+
+// GetAuthToken retrieves the authentication token for the container app specifiec in currentItem
+func (e *ContainerAppExpander) GetAuthToken(ctx context.Context, currentItem *TreeNode) (string, error) {
+	var containerAppNode *TreeNode
+
+	// Walk up the tree looking for the container app node
+	for node := currentItem; node != nil; node = node.Parent {
+		if node.SwaggerResourceType != nil && node.SwaggerResourceType.Endpoint.TemplateURL == "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/containerApps/{containerAppName}" {
+			containerAppNode = node
+			break
+		}
+	}
+	if containerAppNode == nil {
+		return "", fmt.Errorf("failed to find container app node")
+	}
+
+	containerAppEndpoint := containerAppNode.SwaggerResourceType.Endpoint
+	authEndpoint, err := endpoints.GetEndpointInfoFromURL("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/containerApps/{containerAppName}/getAuthToken", containerAppEndpoint.APIVersion)
+	if err != nil {
+		return "", fmt.Errorf("failed to get auth endpoint: %s", err)
+	}
+
+	match := containerAppEndpoint.Match(containerAppNode.ExpandURL)
+	if !match.IsMatch {
+		return "", fmt.Errorf("failed to match container app URL")
+	}
+
+	authURL, err := authEndpoint.BuildURL(match.Values)
+	if err != nil {
+		return "", fmt.Errorf("failed to build auth URL: %s", err)
+	}
+
+	data, err := e.client.DoRequest(ctx, "POST", authURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to get auth token: %s", err)
+	}
+
+	var authTokenResponse ContainerAppAuthTokenResponse
+	err = json.Unmarshal([]byte(data), &authTokenResponse)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse auth token response: %s", err)
+	}
+	return authTokenResponse.Properties.Token, nil
+}
+
+// func (e *ContainerAppExpander) getLogs(ctx context.Context, logStreamEndpoint string, authToken string) (string, error) {
```

Comment:
Think this can be removed now right? As the command handler is doing it

Reply:
Oops - I meant to remove this!